### DISCUSSION TITLE
QL for .ly files

### DISCUSSION THREAD

**Discussion by regmaark** [sbarex/SourceCodeSyntaxHighlight#246] [Q&A]
Is is possible to add .ly files to those that are supported by this wonderful app, please?

This is the UTI info from the Terminal command mdls:

kMDItemContentType     = "com.unknown.ly"
kMDItemContentTypeTree = (
    "com.unknown.ly",
    "public.text",
    "public.data",
    "public.item",
    "public.content",
    "public.plain-text"
)

**Comment by sbarex** [comment]
Which programming language does the ly extension correspond to?

**Comment by ghost** [comment]
The ly extension is for lilypond: https://lilypond.org/

There is a good lilyPond sheet music text editor, it's frescobaldi: https://www.frescobaldi.org/

The frescobaldi's definition of color is here: 

https://github.com/frescobaldi/python-ly/blob/master/ly/colorize.py

**Comment by sbarex** [comment]
I can support the syntax formatting, not displaying the musical score.

**Comment by ghost** [comment]
Lilypond is the equivalent of pdftex for musical scores. The ly file is a text file containing the score code, just like the tex file for pdftex. 
Frescobaldi is the equivalent of texstudio, which opens the ly file, colorizes its syntax and also displays the pdf containing the musical score. Frescobaldi is also maintained by the developers of lilypond, and its syntax colorization can be considered a reference. 
Frescobaldi defines the colors to be used in this python file:
https://github.com/frescobaldi/python-ly/blob/master/ly/colorize.py
Afterwards, if the syntax formatting is too complex, simply displaying the ly files as text files would be a nice touch.

**Comment by sbarex** [comment]
Implemented on release 2.1.21.

**Comment by ghost** [comment]
You're quick, thanks!

**Comment by ghost** [comment]
I haven't thought about it any more, but it would also be great to do the same with files with the ily extension. They're exactly the same as files with the ly extension, the only difference being that you indicate that they can only be included in a ly file and not launched directly.

**Comment by ghost** [comment]
Actually it doesn't work with .ly files, it says that the extension com.apple.tips.TipsAppQuicklook-macos doesn't implement the file preview.

**Comment by sbarex** [comment]
I believe the extension `.ly` is handled by some other generator.

see: https://apple.stackexchange.com/questions/450357/quicklook-for-latex-and-other-text-files-stopped-working-macos-13

**Comment by ghost** [comment]
Yes, I think it's internal programming. 

Could you define, that the .ly and .ily files are seen as .txt files?

Can a generator be manually defined?
